integrationtests/src/test/java/org/axonframework/integrationtests/testsuite/administration/state/mutable/MutableEmployee.java

```diff
@@ -0,0 +1,90 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.integrationtests.testsuite.administration.state.mutable;
+
+import org.axonframework.commandhandling.annotation.CommandHandler;
+import org.axonframework.eventhandling.gateway.EventAppender;
+import org.axonframework.eventsourcing.EventSourcingHandler;
+import org.axonframework.integrationtests.testsuite.administration.AnnotationTestDefinitions;
+import org.axonframework.integrationtests.testsuite.administration.commands.AssignTaskCommand;
+import org.axonframework.integrationtests.testsuite.administration.commands.CreateEmployee;
+import org.axonframework.integrationtests.testsuite.administration.events.EmployeeCreated;
+import org.axonframework.integrationtests.testsuite.administration.events.TaskAssigned;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MutableEmployee extends MutablePerson {
+
+    @AnnotationTestDefinitions.EntityMember
+    private MutableSalaryInformation salary;
+    @AnnotationTestDefinitions.EntityMember
+    private List<MutableTask> taskList = new ArrayList<>();
+
+    @CommandHandler
+    public void handle(CreateEmployee command, EventAppender eventAppender) {
+        if (identifier != null) {
+            throw new IllegalStateException("Employee already created");
+        }
+        eventAppender.append(new EmployeeCreated(
+                command.identifier(),
+                command.lastNames(),
+                command.firstNames(),
+                command.emailAddress(),
+                command.role(),
+                command.initialSalary()
+        ));
+    }
+
+    @CommandHandler
+    public void handle(AssignTaskCommand command, EventAppender eventAppender) {
+        if (taskList.stream().filter(s -> !s.isCompleted()).collect(Collectors.toSet()).size() >= 3) {
+            throw new IllegalStateException("Cannot assign more than 3 tasks to an employee");
+        }
+        eventAppender.append(new TaskAssigned(
+                command.identifier(),
+                command.id(),
+                command.description()
+        ));
+    }
+
+    @EventSourcingHandler
+    public void on(EmployeeCreated event) {
+        this.identifier = event.identifier();
+        this.lastNames = event.lastNames();
+        this.firstNames = event.firstNames();
+        this.emailAddress = event.emailAddress();
+        this.salary = new MutableSalaryInformation(event.initialSalary(), event.role());
+    }
+
+    @EventSourcingHandler
+    public void on(TaskAssigned event) {
+        taskList.add(new MutableTask(event.taskId()));
+    }
+
+    public List<MutableTask> getTaskList() {
+        return taskList;
+    }
+
+    public void setTaskList(
+            List<MutableTask> taskList) {
+        this.taskList = taskList;
+    }
```

Comment:
Does it need to be public? If we can make it private or at-least package-private it may be nice.

Reply:
As it's referenced outside the package, it needs to be public